src/main/java/ch/njol/skript/SkriptCommand.java

```diff
@@ -169,41 +170,55 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
 					Aliases.clear();
 					Aliases.loadAsync().thenRun(() -> reloaded(sender, logHandler, timingLogHandler, "aliases"));
 				} else { // Reloading an individual Script or folder
-					File scriptFile = getScriptFromArgs(sender, args);
-					if (scriptFile == null)
-						return true;
-
-					if (!scriptFile.isDirectory()) {
-						if (ScriptLoader.getDisabledScriptsFilter().accept(scriptFile)) {
-							info(sender, "reload.script disabled", scriptFile.getName().substring(ScriptLoader.DISABLED_SCRIPT_PREFIX_LENGTH), StringUtils.join(args, " ", 1, args.length));
+					File[] scriptFiles;
+					if (args[1].equalsIgnoreCase("lastReloaded")) {
+						if (lastReloaded == null || lastReloaded.length == 0) {
+							error(sender, "reload.no previous reload");
 							return true;
 						}
-
-						reloading(sender, "script", logHandler, scriptFile.getName());
-
-						Script script = ScriptLoader.getScript(scriptFile);
-						if (script != null)
-							ScriptLoader.unloadScript(script);
-						ScriptLoader.loadScripts(scriptFile, OpenCloseable.combine(logHandler, timingLogHandler))
-							.thenAccept(scriptInfo ->
-								reloaded(sender, logHandler, timingLogHandler, "script", scriptFile.getName())
-							);
+						scriptFiles = getScriptsFromArgs(sender, lastReloaded);
+						if (scriptFiles == null || scriptFiles.length == 0)
+							return true;
 					} else {
-						final String fileName = scriptFile.getName();
-						reloading(sender, "scripts in folder", logHandler, fileName);
-						ScriptLoader.unloadScripts(ScriptLoader.getScripts(scriptFile));
-						ScriptLoader.loadScripts(scriptFile, OpenCloseable.combine(logHandler, timingLogHandler))
-							.thenAccept(scriptInfo -> {
-								if (scriptInfo.files == 0) {
-									info(sender, "reload.empty folder", fileName);
-								} else {
-									if (logHandler.numErrors() == 0) {
-										reloaded(sender, logHandler, timingLogHandler, "x scripts in folder success", fileName, scriptInfo.files);
+						scriptFiles = getScriptsFromArgs(sender, args);
+						if (scriptFiles == null || scriptFiles.length == 0)
+							return true;
+						lastReloaded = args;
+					}
+
+					for (File scriptFile : scriptFiles) {
```

Comment:
I think it's better for behavior if this disables everything and then re-enables everything. If the scripts are interdependent, this kind of load order (unload a, load a, unload b, load b) could cause some weird behavior. You should also make sure there aren't duplicates.